presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -240,35 +274,59 @@ public boolean usesLegacyTableLayouts()
     @Override
     public ConnectorTableProperties getTableProperties(ConnectorSession session, ConnectorTableHandle table)
     {
-        return new ConnectorTableProperties();
+        ElasticsearchTableHandle handle = (ElasticsearchTableHandle) table;
+
+        return new ConnectorTableProperties(
+                handle.getConstraint(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                emptyList());
     }
 
     @Override
     public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(ConnectorSession session, ConnectorTableHandle table, Constraint constraint)
     {
         ElasticsearchTableHandle handle = (ElasticsearchTableHandle) table;
 
+        Map<ColumnHandle, Domain> supported = new HashMap<>();
+        Map<ColumnHandle, Domain> unsupported = new HashMap<>();
+        if (constraint.getSummary().getDomains().isPresent()) {
```

Comment:
nit: `constraint.getSummary().getDomains().orElseGet(immutableMap::of)` would let you remove the `if`.

(no change requested)